### PR TITLE
Use JSON markdown blocks in docs

### PR DESCRIPTION
**Pull request by j8r**
The comments aren't supported in JSON, that's why there are shown as red.
An option may be to use an other block code, like YAML (or something else, don't know if there is JSON5).

### CODE REVIEW

**Comment by stgraber** [comment]
Hmm, yeah, the fact that it's not valid json is going to be a problem I suspect.
`jsonc` or `json5` would work on Github but pygment doesn't seem to support either as a format.

`yaml` may do the trick as it's a superset of json, I just don't know how well the coloring will be with that.

**Comment by j8r** [comment]
We can also use `js` instead of `json`, and replace `#` comments to `//`

File: doc/rest-api.md [hunk 1]
```diff
@@ -87,12 +93,14 @@ the body can also be retrieved from the background operation URL.
...
-        "error": "Failure",
-        "error_code": 400,
-        "metadata": {}                      # More details about the error
-    }
+```json
+{
+    "type": "error",
+    "error": "Failure",
+    "error_code": 400,
+    "metadata": {}                      // More details about the error
```
Comment (stgraber) [inline doc/rest-api.md]: Missed this one

File: doc/rest-api.md [hunk 2]
```diff
@@ -1058,24 +1150,30 @@ Return:
...
-    {
-        "name": "new-name"
-        "migration": true
-        "live": "true"
-    }
+```json
+{
+    "name": "new-name"
+    "migration": true
+    "live": "true"
```
Comment (stgraber) [inline doc/rest-api.md]: Should probably fix the syntax on this one (missing commas).

File: doc/rest-api.md [hunk 3]
```diff
@@ -1493,9 +1615,11 @@ Input:
 
 Output:
 
-    {
-        "data": <byte-stream>
-    }
+```json
+{
+    "data": <byte-stream>
```
Comment (stgraber) [inline doc/rest-api.md]: Bad syntax, probably just put it as a string for now.

File: doc/rest-api.md [hunk 4]
```diff
@@ -1520,25 +1644,29 @@ The notification types are:
...
+
+```json
+{
+    "timestamp": "2016-02-17T11:44:28.572721913-05:00",
+    "type": "logging",
+    "metadata": {
+        "context": {
+            "ip": "@",
+            "method": "GET"
+            "url": "/1.0/containers/xen/snapshots",
```
Comment (stgraber) [inline doc/rest-api.md]: Bad syntax due to missing comma

File: doc/rest-api.md [hunk 5]
```diff
@@ -2595,38 +2823,44 @@ Return:
...
-            "pool": "pool2",
-            "name": "vol2",
-            "type": "copy"
-        }
+```json
+{
+    "config": {},
+    "name": "vol1",
+    "type": "custom"
+    "source": {
```
Comment (stgraber) [inline doc/rest-api.md]: Bad syntax due to missing comma

File: doc/rest-api.md [hunk 6]
```diff
@@ -2933,42 +3202,48 @@ Return:
...
+            "value": "/dev/sdb",
+        },
+        {
+            "entity": "network",
+            "name": "lxdbr0",
+            "key": "bridge.external_interfaces",
+            "value": "vlan0",
+        },
+}
+```
```
Comment (stgraber) [inline doc/rest-api.md]: Not sure what's going on around here but github looks rather unhappy.

**Comment by stgraber** [comment]
Switching to json or js has now made github do a better job at validating things, showing a number of issues that we should probably fix as part of this.

File: doc/rest-api.md (hunk 6 above)
Comment (j8r) [reply doc/rest-api.md]: It due to the upper comma `},`

Comment (j8r) [reply doc/rest-api.md]: oh, missing closing bracket `]`

**Comment by stgraber** [comment]
Looks good, thanks!

**Pull request merged**